File: 2710-minimum-operations-to-reduce-an-integer-to-0/minimum-operations-to-reduce-an-integer-to-0.py
class Solution(object):
    def minOperations(self, n):
        """
        :type n: int
        :rtype: int
        """
        binary = bin(n)
        co=0
        res = 0

        while n>0:
            if n&1 == 1:
                co+=1
            elif co>0:
                res +=1
                co = 0 if co==1 else 1
            n>>=1
        if co==1:
            res +=1
        elif co> 1:
            res +=2
        return res




        

File: 2710-minimum-operations-to-reduce-an-integer-to-0/minimum-operations-to-reduce-an-integer-to-0.cpp
class Solution {
public:
    int minOperations(int n) {
        
        int carry = 0;
        int result = 0;
        while(n>0){
            if (n&1){
                carry++;
            }
            else if(carry >0){
                result++;
                carry = carry == 1 ? 0 :1;
            }
            n>>=1;
        }

        if(carry ==1){
            result++;
        }else if (carry > 1){
            result +=2;
        }
        return result;
    }
};